lib/scan_beacon/ble112_device.rb

```diff
@@ -32,6 +33,12 @@ def open
       @file = nil
     end
 
+    def configure_port
+      if RUBY_PLATFORM =~ /linux/
+        system("stty -F #{@port} 115200 -brkint -icrnl -imaxbel -opost -isig -icanon -iexten -echo -echoe -echok -echoctl -echoke;min=0")
+      end
```

Comment:
[Open3](http://ruby-doc.org/stdlib-1.9.3/libdoc/open3/rdoc/Open3.html) is the way to go if you want/need to handle stderr. For some reason `popen` doesn't give you access to that handle. Luckily Open3 is in the standard lib. Avdi has a good [blog post](https://devver.wordpress.com/2009/10/12/ruby-subprocesses-part_3/) about it.